### PR TITLE
Adapt behavior of stats.* cluster setting to match behavior of ES cluster wide settings

### PR DESCRIPTION
**Pull request by matriv**
See trello card for discovery.zen.minimum_master_nodes behavior:
https://trello.com/c/1aUofsJX/1775-fix-bug-with-stats-enabled

### CODE REVIEW

**Comment by matriv** [comment]
@aelesbao then @dobe Please review

**Comment by aelesbao** [comment]
could you remove the commented lines on the diff?

File: sql/src/main/java/io/crate/operation/collect/StatsTables.java [hunk 1]
```diff
@@ -71,9 +71,12 @@
     private final LongAdder activeRequests = new LongAdder();
 
     protected final NodeSettingsService.Listener listener = new NodeSettingListener();
+    private int initialOperationsLogSize;
+    private int initialJobsLogSize;
+    boolean initialIsEnabled;
     volatile int lastOperationsLogSize;
     volatile int lastJobsLogSize;
-    private volatile boolean lastIsEnabled;
+    volatile boolean lastIsEnabled;
```
Comment (aelesbao) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: why did you change visibility?

File: sql/src/main/java/io/crate/operation/collect/StatsTables.java [hunk 2]
```diff
@@ -71,9 +71,12 @@
     private final LongAdder activeRequests = new LongAdder();
 
     protected final NodeSettingsService.Listener listener = new NodeSettingListener();
+    private int initialOperationsLogSize;
+    private int initialJobsLogSize;
+    boolean initialIsEnabled;
```
Comment (aelesbao) [inline sql/src/main/java/io/crate/operation/collect/StatsTables.java]: could this be private?

**Comment by aelesbao** [comment]
through

**Comment by matriv** [comment]
@aelesbao fixup pushed

**Comment by aelesbao** [comment]
@dobe could you do the final review for this one please?

File: sql/src/main/java/io/crate/metadata/settings/Setting.java [hunk 3]
```diff
@@ -43,6 +43,8 @@ public String settingName() {
 
     public abstract E extract(Settings settings);
 
+    public abstract E extract(Settings settings, T defaultValue);
```
Comment (dobe) [inline sql/src/main/java/io/crate/metadata/settings/Setting.java]: @Nullable is missing
pls document this behaviour

File: sql/src/main/java/io/crate/metadata/settings/TimeSetting.java [hunk 4]
```diff
@@ -48,14 +48,23 @@ public DataType dataType() {
...
+    }
+
+    @Override
+    public String extract(Settings settings, TimeValue defaultValue) {
+        return extractTimeValue(settings, defaultValue).toString();
     }
 
     public long extractMillis(Settings settings) {
-        return extractTimeValue(settings).millis();
+        return extractTimeValue(settings, null).millis();
```
Comment (dobe) [inline sql/src/main/java/io/crate/metadata/settings/TimeSetting.java]: could call without second arg

File: sql/src/main/java/io/crate/metadata/settings/Setting.java (hunk 3 above)
Comment (matriv) [inline sql/src/main/java/io/crate/metadata/settings/Setting.java]: Document it where?

File: CHANGES.txt [hunk 5]
```diff
@@ -5,9 +5,9 @@ Changes for Crate
 Unreleased
 ==========
 
- - Adapted behavior of stats.enabled, stats.jobs_log_size and stats.operations_log_size
-   settings to match behavior of Elasticsearch cluster wide settings.
-   Example: discovery.zen.minimum_master_nodes
+ - Adapted behaviour of stats.enabled, stats.jobs_log_size and stats.operations_log_size
+   settings to match behaviour of other cluster wide settings
+   like for example minimum_master_nodes.
```
Comment (dobe) [inline CHANGES.txt]: pls fix linebreaks

**Comment by dobe** [comment]
otherwise ack
